lib/publishing_api_finder_publisher.rb

```diff
@@ -0,0 +1,70 @@
+require "gds_api/publishing_api_v2"
+
+class PublishingApiFinderPublisher
+  def initialize(finders, logger: Logger.new(STDOUT))
+    @finders = finders
+    @logger = logger
+  end
+
+  def call
+    finders.map do |finder|
+      if pre_production?(finder)
+        publish(finder)
+      else
+        logger.info("Not publishing #{finder[:file]['name']} because it is pre_production")
+      end
+    end
+  end
+
+  def publishing_api
+    @publishing_api ||= GdsApi::PublishingApiV2.new(
```

Comment:
This method is only being called by private methods, so should probably be private itself.